### ISSUE TITLE
Metacello integration is broken in 0.6.7

### ISSUE THREAD

**Issue by guillep** [pharo-vcs/iceberg#577]
Metacello integration is supposed to try cloning through ssh and then through https if ssh connection is not possible. 

MCGitBasedNetworkRepository checks an authentication error and gives:

```smalltalk
 [ try clone ]
        on: IceAuthenticationError
        do: [ :e | try https ]
```

This relies on the clone failing with an `IceAuthenticationError`. This was apparently working before but not it seems that the ssh credential callback is overriding the error with a EUSER status and thus we do not get an IceAuthenticationError anymore in:

```smalltalk
visitERROR: anError
	(anError object = LGitErrorTypeEnum giterr_ssh)
		ifTrue: [ anError resignalAs: IceAuthenticationError new ].
	self visitGenericError: anError
```

The bug seems to be in the following line:

```smalltalk
createFrom: provider username: username onto: lgitCredentials
	| credentials |
	
	[  
		credentials := self new.
		provider provideSshAgentCredentialsTo: credentials.	
		credentials createWith: username onto: lgitCredentials.
		^ 0 ]
	on: LGitNoCredentialsProvided do: [ 
====>>> LGitError setError: 'No ssh-agent suitable credentials found.'.
		^ -1 ]
```

Removing that line restores the https fallback and the nice ssh error message.

**Comment by estebanlm** [comment]
this is fixed in 0.7, we need to backport it.

**Comment by estebanlm** [comment]
in fact... ois fixed in my branch. 
I need to adapt it yo 0.6.7 and commit :)
